src/main/java/com/quorum/gauge/services/RaftService.java

```diff
@@ -44,6 +48,44 @@
         });
     }
 
+    public Observable<RaftCluster> getCluster(QuorumNode node) {
+        Request<String, RaftCluster> request = new Request<>(
+                "raft_cluster",
+                null,
+                connectionFactory().getWeb3jService(node),
+                RaftCluster.class);
+        return request.observable();
+    }
+
+    /**
+     * Retrieve the enode for the raft leader and look it up in the
+     * configured nodes to convert it into a QuorumNode identity.
+     */
+    public QuorumNode getLeader(QuorumNode node) {
+        Request<String, RaftLeader> request = new Request<>(
+                "raft_leader",
+                null,
+                connectionFactory().getWeb3jService(node),
+                RaftLeader.class);
+        RaftLeader response = request.observable().toBlocking().first();
+        String leaderEnode = response.getResult();
+
+        Map<QuorumNode, QuorumNetworkProperty.Node> nodes = connectionFactory().getNetworkProperty().getNodes();
+        QuorumNode leaderNodeId = null;
+        for (QuorumNode nodeId : nodes.keySet()) {
+            QuorumNetworkProperty.Node nodeProperties = nodes.get(nodeId);
+            if (nodeProperties.getEnode().equals(leaderEnode)) {
```

Comment:
I forgot to commit the enode details in the yaml, so I've added that now. I've also added a check for the null enode case.
The enode details will also be needed to be added to application-local.sample.yml. (I don't know what their values are.)

I had a look at using admin_nodeInfo, but it seems to be incomplete in web3j, i.e. the response object NodeInfo is not defined. Also, if I did manage to use that then I'd have to query every node in the network for their enode and then compare that against the leader enode, which seemed very wasteful.
Let me know what you think.

Reply:
I would try to avoid hardcoding values which can be obtained via API, it makes tests less coupling with static data.

You can call `admin_nodeInfo` API with vanilla `Request`, there are some examples in the code already using this approach (e.g.: `IstanbulService` or `ContractService#createClientReceiptContractAsync`)

I would not worry about the querying the network for enode values as it's fast and we can parallelize the calls with `Observable.zip()`